api/admin/controller.py

```diff
@@ -2323,6 +2325,109 @@ def _sitewide_settings_controller(self, configuration_object):
         setting.value = value
         return Response(unicode(setting.key), 200)
 
+    def _get_protocols(self, provider_apis):
+        protocols = self._get_integration_protocols(provider_apis, protocol_name_attr="NAME")
+        protocols.append(dict(name=ExternalIntegration.MANUAL,
+                              label=_("Manual import"),
+                              description=_("Books will be manually added to the circulation manager, not imported automatically through a protocol."),
+                              settings=[],
+                              ))
+
+        return protocols
+
+    def _get_prior_test_results(self, collection, newCollection):
+        """This helper function returns previous self test results for a given
+        collection if it has a protocol. The `collection` is the collection found
+        in the database and the `newCollection` is the collection being constructed.
```

Comment:
I don't understand this comment. Why is a new collection being constructed?

Reply:
I'm not too happy with how I worked this out. I'll try to see if I can do it differently, but passing just the protocol class is a start.